maven-plugin-plugin/src/main/java/org/apache/maven/plugin/plugin/AbstractGeneratorMojo.java

```diff
@@ -81,28 +80,24 @@ public void execute() throws MojoExecutionException {
             return;
         }
 
-        String defaultGoalPrefix = getDefaultGoalPrefix(project);
-
-        if (goalPrefix == null) {
-            goalPrefix = defaultGoalPrefix;
-        } else if (!goalPrefix.equals(defaultGoalPrefix)) {
-            getLog().warn(LS + LS + "Goal prefix is specified as: '" + goalPrefix + "'. "
-                    + "Maven currently expects it to be '" + defaultGoalPrefix + "'." + LS);
+        if (goalPrefix == null || goalPrefix.isEmpty()) {
+            goalPrefix = getDefaultGoalPrefix(project);
+        }
+        if (goalPrefix == null || goalPrefix.isEmpty()) {
+            throw new MojoExecutionException("You need to specify a goalPrefix as it can not be correctly computed");
         }
 
         generate();
     }
 
     static String getDefaultGoalPrefix(MavenProject project) {
-        String defaultGoalPrefix;
-        if ("maven-plugin-report-plugin".equalsIgnoreCase(project.getArtifactId())) {
-            defaultGoalPrefix = "plugin-report";
-        } else if ("maven-plugin".equalsIgnoreCase(project.getArtifactId())) {
-            defaultGoalPrefix =
-                    project.getGroupId().substring(project.getGroupId().lastIndexOf('.') + 1);
+        String artifactId = project.getArtifactId();
+        if (artifactId.endsWith("-maven-plugin")) {
+            return artifactId.substring(0, artifactId.length() - "-maven-plugin".length());
+        } else if (artifactId.startsWith("maven-") && artifactId.endsWith("-plugin")) {
+            return artifactId.substring("maven-".length(), artifactId.length() - "-plugin".length());
         } else {
-            defaultGoalPrefix = PluginDescriptor.getGoalPrefixFromArtifactId(project.getArtifactId());
```

Comment:
Should this imply that this static method should be deprecated?

Reply:
In 4.0 ? I'm currently working on cleaning the maven-plugin-api module, i.e. getting rid of plexus in the API and generating the model using the mdo file, etc... So the whole package will be somewhat deprecated at some point.  Not sure we should deprecate a single method...

Reply:
Alright, then not worth it.